src/index.js

```diff
@@ -58,13 +58,15 @@ export class PostmsgTransport {
     if(!e.data || typeof(e.data) !== 'string') {
       return;
     }
+    let data;
     try {
-      const { key, event } = JSON.parse(e.data);
-      if (key === this._key) {
-        this._handler(event);
-      }
+      data = JSON.parse(e.data);
     } catch (e) {
       return null;
     }
+    const { key, event } = data;
```

Comment:
I don't think it'll work. "null" is a valid JSON string so the data variable will be null if this comes up. Post Message is global so better add a check for this here.

```
JSON.parse('null')
=> null
```